From 852eda8889f258294720e3a671299d19a662df2f Mon Sep 17 00:00:00 2001
From: Adrian DC <radian.dc@gmail.com>
Date: Sun, 31 Dec 2017 13:12:05 +0100
Subject: [PATCH] stagefright: ACodec: Resolve empty vendor parameters usage

 * On devices where vendorKeys is empty, the status was left
    as kExtensionsUnchecked and would block getVendorParameters
    on a wrong loop of empty vendor keys

 * Instead, set the status to kExtensionsNone to validate
    the check and the empty vendor keys, matching the old
    behaviour without vendor keys in getPortFormat

Change-Id: Ic3bdc9454c6af8e5d41ab901d4156ccb62281974
Signed-off-by: Adrian DC <radian.dc@gmail.com>
(cherry picked from commit b96ace1d3df35e7c5e1d73dc9362f6c0d890bc10)
(cherry picked from commit 0a01c0c5f027e94d961ac3a7d0e9ab39d78abe38)
---
 media/libstagefright/ACodec.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/media/libstagefright/ACodec.cpp b/media/libstagefright/ACodec.cpp
index 2c8d8de226..2b7e198a58 100644
--- a/media/libstagefright/ACodec.cpp
+++ b/media/libstagefright/ACodec.cpp
@@ -8151,6 +8151,7 @@ status_t ACodec::setVendorParameters(const sp<AMessage> &params) {
     // don't bother component if we don't have vendor extensions as they may not have implemented
     // the android vendor extension support, which will lead to unnecessary OMX failure logs.
     if (vendorKeys.empty()) {
+        mVendorExtensionsStatus = kExtensionsNone;
         return OK;
     }
 
